consensus/consensus.go

```diff
@@ -80,96 +79,89 @@ type Engine interface {
 
 	// Finalize runs any post-transaction state modifications (e.g. block rewards)
 	// but does not assemble the block.
 	//
 	// Note: The block header and state database might be updated to reflect any
 	// consensus rules that happen at finalization (e.g. block rewards).
-	Finalize(chain ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction,
-		uncles []*types.Header)
+	Finalize(chain ChainHeaderReader, header *types.WorkObject, state *state.StateDB)
 
 	// FinalizeAndAssemble runs any post-transaction state modifications (e.g. block
 	// rewards) and assembles the final block.
 	//
 	// Note: The block header and state database might be updated to reflect any
 	// consensus rules that happen at finalization (e.g. block rewards).
-	FinalizeAndAssemble(chain ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header, etxs []*types.Transaction, manifest types.BlockManifest, receipts []*types.Receipt) (*types.Block, error)
+	FinalizeAndAssemble(chain ChainHeaderReader, woHeader *types.WorkObject, state *state.StateDB, txs []*types.Transaction, uncles []*types.WorkObject, etxs []*types.Transaction, subManifest types.BlockManifest, receipts []*types.Receipt) (*types.WorkObject, error)
 
 	// Seal generates a new sealing request for the given input block and pushes
 	// the result into the given channel.
 	//
 	// Note, the method returns immediately and will send the result async. More
 	// than one result may also be returned depending on the consensus algorithm.
-	Seal(header *types.Header, results chan<- *types.Header, stop <-chan struct{}) error
+	Seal(header *types.WorkObject, results chan<- *types.WorkObject, stop <-chan struct{}) error
 
 	// CalcDifficulty is the difficulty adjustment algorithm. It returns the difficulty
 	// that a new block should have.
-	CalcDifficulty(chain ChainHeaderReader, parent *types.Header) *big.Int
+	CalcDifficulty(chain ChainHeaderReader, parent *types.WorkObject) *big.Int
 
 	// IsDomCoincident returns true if this block satisfies the difficulty order
 	// of a dominant chain. If this node does not have a dominant chain (i.e.
 	// if this is a prime node), then the function will always return false.
 	//
 	// Importantly, this check does NOT mean the block is canonical in the
 	// dominant chain, or even that the claimed dominant difficulty is valid.
-	IsDomCoincident(chain ChainHeaderReader, header *types.Header) bool
+	IsDomCoincident(chain ChainHeaderReader, header *types.WorkObject) bool
 
 	// VerifySeal computes the PowHash and checks if work meets the difficulty
 	// requirement specified in header
-	VerifySeal(header *types.Header) (common.Hash, error)
-
-	// APIs returns the RPC APIs this consensus engine provides.
-	APIs(chain ChainHeaderReader) []rpc.API
-
-	// Close terminates any background threads maintained by the consensus engine.
-	Close() error
```

Comment:
Why did you get rid of `Close()`? Are there no longer any background processes? This seems like a weird change for a workobject PR

Reply:
No, we dont have any background processes for the consensus engines, everything we use is just client side one off functions